### ISSUE TITLE
Pre-signing doesn't work for multipart uploads

### ISSUE THREAD

**Issue by archseer** [ex-aws/ex_aws_s3#36]
I'm trying to re-create the AWS JS SDK code from here: https://github.com/transloadit/uppy/blob/7a137de07989f161bd16f6b56696bba9523da66e/packages/%40uppy/companion/src/server/controllers/s3.js#L178-L185

```elixir
  def initiate_upload(key, type) do
    config = Application.fetch_env!(:ex_aws, :s3)

    response =
      ExAws.S3.initiate_multipart_upload(config[:bucket], key, [
        content_type: type,
        expires: 3600 * 15,
        acl: :public_read,
      ])
      |> ExAws.request!()
    {:ok, response.body}
  end

  def prepare_upload_part(key, upload_id, part_number) do
    config = Application.fetch_env!(:ex_aws, :s3)

    ExAws.S3.presigned_url(ExAws.Config.new(:s3), :put, config[:bucket], key, [
      expires_in: 3600 * 15,
      query_params: [
        partNumber: part_number,
        uploadId: upload_id,
      ]
    ])
  end
```

Generating a presigned URL for a part of a multipart upload:

```elixir
{:ok, response} = initiate_upload("text.txt", "text/plain")
{:ok, url} = prepare_upload_part("text.txt", response.upload_id, 1)
```

Then trying to upload a part to that URL via the browser:
```
curl '<url>' -X PUT -H 'User-Agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10.13; rv:64.0) Gecko/20100101 Firefox/64.0' -H 'Accept: */*' -H 'Accept-Language: en-US,en;q=0.5' --compressed -H 'Referer: http://localhost:8080/about' -H 'Origin: http://localhost:8080' -H 'DNT: 1' -H 'Connection: keep-alive' --data $'hello world\n'
```

Fails with `SignatureDoesNotMatch`.

Removing the `query_params` works, but then we're just doing a regular upload via PUT instead of a multipart upload (the part parameters are missing).

**Comment by sorentwo** [comment]
@archseer It has been quite a while. Did you ever resolve this issue?

I've just finished setting up a multipart upload flow with presigned urls and I didn't encounter any trouble. One thing I noticed is that the type signature for the `query_params` is `[{binary(), binary()}]`, where your usage was `[{atom(), binary() | integer()}]`. The query params need to have both binary keys and values.

**Comment by kerryjj** [comment]
Looks like this issue has been fixed and can be closed? This is the comment from the merged PR above.

#36 - fixed by ex-aws/ex_aws#540 and this PR (see comment)

**Comment by archseer** [comment]
I think #79 fixed it, I haven't used this library in a while.
